### PR TITLE
ci: consistent formatting across packages

### PR DESCRIPTION
**Pull request by benelan**
## Summary

- Add linting npm scripts to additional packages.
- Run `prettier` before `markdownlint` in `lint:md` npm scripts, because `prettier` was overriding some styling.
- Switch italics/bold style to asterisk instead of underscore, because `prettier` and `conventional-changelog` both use astericks. The formatting was getting out of sync in the changelog during next releases (see link below for an example). This would result in unrelated diffs in PRs.
- Lint code.

https://github.com/Esri/calcite-design-system/blob/c9aabd27c815a309a2756971e38c5903161bbb7b/packages/calcite-components-react/CHANGELOG.md?plain=1#L46-L55

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍

**Comment by benelan** [comment]
@driskull - Re-requested review because the purpose of the PR changed. The changelog diffs in your PR was a CI issue, which this fixes.

**Comment by driskull** [comment]
Thanks for the heads up! I wasn't sure how it happened :)

File: .markdownlintrc [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "code-fence-style": { "style": "backtick" },
   "heading-style": { "style": "atx" },
   "no-duplicate-heading": { "siblings_only": true },
-  "emphasis-style": { "style": "underscore" },
-  "strong-style": { "style": "underscore" },
+  "emphasis-style": { "style": "asterisk" },
```
Comment (driskull) [inline .markdownlintrc:10]: *️⃣ 👍

**Review by driskull — APPROVED** [review]
Nice! Consistency FTW :godmode:

**Pull request merged**